Snowplow/SPUtilities.m

```diff
@@ -170,12 +178,20 @@ + (NSString *) getNetworkType {
 }
 
 + (NSString *) getNetworkTechnology {
-    NSString * netTech = nil;
 #if SNOWPLOW_TARGET_IOS
-    CTTelephonyNetworkInfo *netInfo = [[CTTelephonyNetworkInfo alloc] init];
-    netTech = [netInfo currentRadioAccessTechnology];
+    CTTelephonyNetworkInfo *networkInfo = [CTTelephonyNetworkInfo new];
+    if (@available(iOS 12.1, *)) {
+        // `serviceCurrentRadioAccessTechnology` has a bug in the iOS 12.0 so we use it from iOS 12.1
+        NSDictionary<NSString *, NSString *> *services = [networkInfo serviceCurrentRadioAccessTechnology];
+        NSArray<NSString *> *technologies = services.allValues;
+        // From iOS 12, iPhones with eSIMs can return multiple radio access technologies.
+        // We can't prefer anyone of them so we track the first reported.
+        return technologies.firstObject;
```

Comment:
The documentation (https://developer.apple.com/documentation/coretelephony/cttelephonynetworkinfo/3024511-servicesubscribercellularprovide) doesn't provide any reassurance that the first radioTechnology is linked to the same service as the first CTCarrier. However, the dictionary does use the same key between CellularProvider and RadioAccessTechnology. Whilst we can't pick from the two carrier options, would it be better to ensure we're tracking the matching radio technology for the _first_ carrier?

Reply:
Good point 👍